kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/row/SimpleBigQueryWriter.java

```diff
@@ -52,16 +52,17 @@ public SimpleBigQueryWriter(BigQuery bigQuery, int retry, long retryWait) {
   }
 
   /**
-   * Sends the request to BigQuery, and throws an exception if any errors occur as a result of doing
-   * so.
+   * Sends the request to BigQuery, and return a map of insertErrors in case of partial failure.
```

Comment:
technically we may return insertErrors for kinds of errors other than partial failures. Trying to data into a table where the schema doesn't match, for example.

Reply:
No. Anything that has a non-200 error code will be an exception. (This is because the bigquery client is just a thin wrapper over a rest interface.)

Take a look at the "Success HTTP response codes" section in that troubleshooting page to see the types of "errors" that will not throw but instead return an insertResponse with errors.

This line seems particularly useful:
> Rows that did not have a schema mismatch will have an error with the reason property set to stopped, and can be re-sent as-is.

Reply:
Now I am clear! Thanks!